packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -612,163 +612,80 @@ contract OPContractsManagerUpgrader is OPContractsManagerBase {
-                    address(ethLockbox),
-                    impls.ethLockboxImpl,
-                    abi.encodeCall(IETHLockbox.initialize, (_opChainConfigs[i].systemConfigProxy, portals))
-                );
-
-                // Migrate liquidity from the OptimismPortal to the ETHLockbox.
-                optimismPortal.migrateLiquidity();
-            }
-
-            // Separate context to avoid stack too deep.
-            {
-                // Grab chain addresses here. We need to do this after the SystemConfig upgrade or
-                // the addresses will be incorrect.
-                ISystemConfig.Addresses memory opChainAddrs = _opChainConfigs[i].systemConfigProxy.getAddresses();
+            // Upgrade the OptimismPortal (no upgrade call).
+            upgradeTo(
+                _opChainConfigs[i].proxyAdmin,
+                address(_opChainConfigs[i].systemConfigProxy.optimismPortal()),
+                address(impls.optimismPortalImpl)
+            );
 
-                // Upgrade the L1CrossDomainMessenger contract.
-                upgradeToAndCall(
-                    _opChainConfigs[i].proxyAdmin,
-                    address(IL1CrossDomainMessenger(opChainAddrs.l1CrossDomainMessenger)),
-                    impls.l1CrossDomainMessengerImpl,
-                    abi.encodeCall(IL1CrossDomainMessenger.upgrade, (_opChainConfigs[i].systemConfigProxy))
-                );
+            // Upgrade the L1CrossDomainMessenger (no upgrade call).
+            upgradeTo(
+                _opChainConfigs[i].proxyAdmin,
+                address(opChainAddrs.l1CrossDomainMessenger),
+                address(impls.l1CrossDomainMessengerImpl)
+            );
 
-                // Upgrade the L1StandardBridge contract.
-                upgradeToAndCall(
-                    _opChainConfigs[i].proxyAdmin,
-                    address(IL1StandardBridge(payable(opChainAddrs.l1StandardBridge))),
-                    impls.l1StandardBridgeImpl,
-                    abi.encodeCall(IL1StandardBridge.upgrade, (_opChainConfigs[i].systemConfigProxy))
-                );
+            // Upgrade the L1StandardBridge (no upgrade call).
+            upgradeTo(
+                _opChainConfigs[i].proxyAdmin,
+                address(opChainAddrs.l1StandardBridge),
+                address(impls.l1StandardBridgeImpl)
+            );
 
-                // Upgrade the L1ERC721Bridge contract.
-                upgradeToAndCall(
-                    _opChainConfigs[i].proxyAdmin,
-                    address(IL1ERC721Bridge(opChainAddrs.l1ERC721Bridge)),
-                    impls.l1ERC721BridgeImpl,
-                    abi.encodeCall(IL1ERC721Bridge.upgrade, (_opChainConfigs[i].systemConfigProxy))
-                );
-            }
+            // Upgrade the L1ERC721Bridge (no upgrade call).
+            upgradeTo(
+                _opChainConfigs[i].proxyAdmin, address(opChainAddrs.l1ERC721Bridge), address(impls.l1ERC721BridgeImpl)
+            );
```

Comment:
Is there anything that we can do in this block of `upgradeTo` calls to ensure that all the necessary contracts are upgraded and nothing is missed?

Reply:
There's a check script that already runs that verifies that any contract that has an upgrade function is being called here. See opcm-upgrade-checks in the scripts folder.